test/mock_batch_test.dart

```diff
@@ -0,0 +1,73 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:cloud_firestore_mocks/cloud_firestore_mocks.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() async {
+  TestWidgetsFlutterBinding.ensureInitialized();
+  String users = "users";
+  MockFirestoreInstance firestore = MockFirestoreInstance();
+
+  Map<String, dynamic> userDocumentMock(String userId) {
+    Map<String, dynamic> data = {
+      "id": userId,
+      "updatedAt": DateTime.now(),
+      "createdAt": DateTime.now(),
+    };
+    return data;
+  }
+
+  Future<List<Map<String, dynamic>>> createUsers({int times}) async {
+    List<Map<String, dynamic>> userDataList = [];
+    for (int i = 1; i <= times; ++i) {
+      String userId = i.toString();
+      userDataList.add(userDocumentMock(userId));
+    }
+    return userDataList;
+  }
+
+  setUp(() async {});
+  tearDown(() async {});
+
+  group("batch", () {
+    test("succees when create 500 documents at once.", () async {
+      List<Map<String, dynamic>> userDataList = await createUsers(times: 500);
```

Comment:
When you convert createUsers to be synchronous, you will then be able to remove `await`.